picky-server/src/http/controller.rs

```diff
@@ -229,6 +232,88 @@ impl ServerController {
             }
         }
     }
+
+    #[post("/timestamp")]
+    async fn authenticode_timestamp(&self, req: Request) -> Result<ResponseBuilder, StatusCode> {
+        let req = req.load_body().await.bad_request()?;
+
+        let mut body = req.body().to_vec();
+        body.retain(|&x| x != b'\n' && x != b'\r' && x != b'\0'); // Removing CRLF entries
+        
+        let der = base64::decode(body)
+            .map_err(|e| format!("base64 failed to decode timestamp request body: {}", e))
+            .internal_error()?;
+        let timestamp_request = TimestampRequest::from_der(&der).bad_request()?;
+
+        let config = self.read_conf().await;
+
+        let intermediate_name = format!("{} Authority", config.realm);
+        let intermediate_hash = self
+            .storage
+            .get_addressing_hash_by_name(&intermediate_name)
+            .await
+            .map_err(|e| format!("couldn't fetch intermediate cert: {}", e))
+            .internal_error()?;
+
+        let intermediate_cert_der = self
+            .storage
+            .get_cert_by_addressing_hash(&intermediate_hash)
+            .await
+            .map_err(|e| format!("couldn't get intermediate cert der: {}", e))
+            .internal_error()?;
+
+        let intermediate_cert = Cert::from_der(&intermediate_cert_der)
+            .map_err(|e| format!("couldn't deserialize intermediate cert: {}", e))
+            .internal_error()?;
+
+        let intermediate_pk_der = self
+            .storage
+            .get_key_by_addressing_hash(&intermediate_hash)
+            .await
+            .map_err(|e| format!("couldn't fetch intermediate private key: {}", e))
+            .internal_error()?;
+
+        let intermediate_pk = Picky::parse_pk_from_magic_der(&intermediate_pk_der)
+            .map_err(|e| e.to_string())
+            .internal_error()?;
+
+        let digest = timestamp_request.digest();
+        let picky_server_hash = config.signing_algorithm.inner_hash_algo();
+
+        let attributes = vec![
+            Attribute::new_content_type_pkcs7(),
+            Attribute::new_signing_time(UTCDate::now().into()),
+            Attribute::new_message_digest(picky_server_hash.digest(digest)),
+        ];
+
+        let authenticode_signature = AuthenticodeSignatureBuilder::new()
+            .digest_algorithm(picky_server_hash)
+            .signing_key(&intermediate_pk)
+            .content_info(timestamp_request.content().clone())
```

Comment:
Maybe add `into_content` method such that:
```suggestion
            .content_info(timestamp_request.into_content())
```